modules/accumulo/src/main/java/org/apache/fluo/accumulo/util/ByteArrayUtil.java

```diff
@@ -98,21 +96,59 @@ public static long decodeLong(byte[] ba) {
    * @param listOfBytes Bytes objects to concatenate
    * @return Bytes
    */
-  public static final byte[] concat(Bytes... listOfBytes) {
-    try {
-      // TODO calculate exact array size needed
-      ByteArrayOutputStream baos = new ByteArrayOutputStream();
-      DataOutputStream dos = new DataOutputStream(baos);
-
-      for (Bytes b : listOfBytes) {
-        WritableUtils.writeVInt(dos, b.length());
-        b.writeTo(dos);
-      }
+  public static byte[] concat(Bytes... listOfBytes) {
+    int offset = 0;
+    int counter = 0;
 
-      dos.close();
-      return baos.toByteArray();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+    for (Bytes b : listOfBytes) {
+      counter += b.length() + 1;
```

Comment:
The +1 always space to store the number from `writeVint` in `data[]` 


```
Output of data[]:

4 <- vInt
115
116
114
49
7 <-vInt
115
116
114
105
110
103
50
2 <- vInt
115
51
```
Wouldn't I still need the +1? with the implementation of vLen(int i)?
```
for (Bytes b : listOfBytes) {
      counter += vLen(b.length()) + 1;
}
```
I'm probably just not understanding it.